File: SmartOrderRouting/src/main/java/com/ab/entities/Instrument.java
package com.ab.entities;

import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.Table;

import lombok.Data;



@Entity
@Table(name="Instruments")
@Data
public class Instrument {

	@Id
	private String code;

	private String company;
    private double price;
	private int volume;
	private double open;
	private double high;
	private double low;
	private double priceChange;

	public Instrument(String code, String company, double price, int volume, double open, double high, double low, double priceChange) {
		this.code = code;
		this.company = company;
		this.price = price;
		this.volume = volume;
		this.open = open;
		this.high = high;
		this.low = low;
		this.priceChange = priceChange;
	}

	public Instrument(){

	}
	
	
}

File: angularclient/src/app/models/Instrument.ts

export class Instrument {

	public code: String;
	public company: String;
	public price: number;
	public volume: number;
	public open: number;
	public high: number;
	public low: number;
	public priceChange: number;

    // TODO set types
	constructor(code: String, company: String, price: number, volume: number, open: number, high: number, 
        low: number, priceChange: number) {
		this.code = code;
		this.company = company;
		this.price = price;
		this.volume = volume;
		this.open = open;
		this.high = high;
		this.low = low;
		this.priceChange = priceChange;
	}
	
}
